### PR TITLE
New FPGA Code Sample "ac_int"

### PR DESCRIPTION
**Pull request by shuoniu-intel**
# Adding a New Sample "ac_int"

## Description

This FPGA tutorial shows how to use the `ac_int` data type with some simple examples. This data type can be used in place of native integer types to generate area efficient and optimized designs for the FPGA.

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): Mike Tucker

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object

Review
- [x] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [x] Review readme with Tom Lenth(@tomlenth) and/or Project Manager

### CODE REVIEW

**Review by mdbtucker — APPROVED** [review]
Already reviewed this branch in Shuo's fork.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+#endif
+
+  bool passed = true;
+
+  try {
+    queue q(device_selector, dpc_common::exception_handler);
+
+    constexpr int kVal1 = 1000, kVal2 = 2;
+
+    // Kernel `BasicOpsInt` contains native `int` type addition, multiplication,
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp:147]: Should this kind of comment be repeated or emphasized in the declaration of these functions?   In general the ratio of comments to code in the example is very low.    You don't want comments that repeat what the code says, but you do want comments that explain the purpose of the code that is close by when you read the code.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    // Kernel `BitAccess` demonstrates bit access with bit select operator `[]`
+    // and bit slice write operation `set_slc`. Note: An `ac_int` must be
+    // initialized before being access by bit select operator `[]` and bit slice
+    // operations `slc` and `set_slc`, otherwise it is undefined behavior and
+    // will give you unexpected results.
+    {
+      MyInt14 input = kVal1;
+      MyInt14 output = TestBitAccess(q, input);
+
+      constexpr int golden = 0b001111101111;
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp:207]: This is a case where you may want to use kGolden, as you are giving a name to a specific magic number.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp (hunk 2 above)
Comment (shuoniu-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp:207]: Will update.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp (hunk 1 above)
Comment (shuoniu-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/ac_int.cpp:147]: I feel I can remove these chunks of comments in `main()`, and add some simplified comments with `Test*` function above.

**Review by tomlenth — APPROVED** [review]

**Review by mdbtucker — APPROVED** [review]

**Review by pmpeter1 — APPROVED** [review]

**Comment by mdbtucker** [comment]
Windows CI tests not running for some reason (https://jira.devtools.intel.com/projects/REIE/issues/REIE-1347).  Will wait a day hoping for a resolution, if still unresolved I'll merge the PR without the Windows CI tests.

**Pull request merged**